### ISSUE TITLE
Formatting styled component for nth-child() is wrong 

### ISSUE THREAD

**Issue by wooxudong** [prettier/prettier#5961]
**Input:**
```jsx
const Steps = styled.div`
  @media (min-width: ${breakpoints.lg}) {
    ${Step} {
      margin-bottom: 90px;
    }

    ${Step}:nth-child(odd) {
      ${StepItemDescription} {
        grid-row: 1;
        grid-column: 3 / span 3;
      }
      ${Image} {
        grid-row: 1;
        grid-column: 7 / span 6;
      }
    }

    ${Step}:nth-child(even) {
      ${Image} {
        grid-row: 1;
        grid-column: 3 / span 6;
      }
      ${StepItemDescription} {
        grid-row: 1;
        grid-column: 10 / span 3;
      }
    }
  }
`;
```

**Output:**
```jsx
const Steps = styled.div`
  @media (min-width: ${breakpoints.lg}) {
    ${Step} {
      margin-bottom: 90px;
    }

    ${Step}:nth-child (odd) {
      ${StepItemDescription} {
        grid-row: 1;
        grid-column: 3 / span 3;
      }
      ${Image} {
        grid-row: 1;
        grid-column: 7 / span 6;
      }
    }

    ${Step}:nth-child (even) {
      ${Image} {
        grid-row: 1;
        grid-column: 3 / span 6;
      }
      ${StepItemDescription} {
        grid-row: 1;
        grid-column: 10 / span 3;
      }
    }
  }
`;
```

**Expected behavior:**
Notice the space added between the `nth-child (odd)`, this would not make the nth-child selector work. I am not sure how I can use prettier but still make this work.

**Comment by fisker** [comment]
minimum reproduction

**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEBleAHNTAXkwwE8AbOAEwDpqBLANwAMAdKTTAAQFsaGAQ0wAKXgygBaAO4NqMABZJMARjwAPAJSZgHLlwAkwXHDwBfJLAWSwChhWoiI1atuBm9mD1BYBuEAA0IBB4MAzoyKCCAE7RENIACjEIaMgggkwQcoEgAEbRgmAA1nAw2HiFEgDmyDDRAK5wQQowvBQA6nbwaBVgcNgpDGFMQ2RpYGipQRJocNEwCQVVvILIAGaCFLNBAFZo6gBCBcWl2IL8ADIScOub2yB76tjVVACK9RDwt1tNIBXRs2iaVyglycAoOTw0QkMHackUyAAHAAGIJQiCzdoFPBpKFwQFMG5BACOH3gixCqRQgjQkigcBoNBy0TgpIYLMWgmWqyQGx+QVm4lqDV+aBecHenxuvLuvxgoLh8gUyAATEE6oJ7NUAMIQXgrNJQaBEkD1WYAFVBVL5szMZiAA)
```sh
--parser babel
```

**Input:**
```jsx
const Steps = styled.div`
  @media (min-width: 1px) {
    ${Step}:nth-child(odd) {}
  }
`;
```

**Output:**
```jsx
const Steps = styled.div`
  @media (min-width: 1px) {
    ${Step}:nth-child (odd) {
    }
  }
`;

```

**Comment by thorn0** [comment]
Fixed by #7883. Needs tests.
